core/warning/warn_spec.rb

```diff
@@ -51,6 +51,17 @@ def Warning.warn(msg)
     end
   end
 
+  ruby_version_is '2.7' do
+    it "emits and suppresses warnings for :deprecated" do
+      ruby_exe(%{Warning[:deprecated] = true; $; = ""}, args: "2>&1").should =~ /is deprecated/
+      ruby_exe(%{Warning[:deprecated] = false; $; = ""}, args: "2>&1").should == ""
+    end
+
+    it "emits and suppresses warnings for :experimental" do
+      ruby_exe(%{Warning[:experimental] = true; 0 in a}, args: "2>&1").should =~ /is experimental/
+      ruby_exe(%{Warning[:experimental] = false; 0 in a}, args: "2>&1").should == ""
+    end
+  end
```

Comment:
These specs should be in `core/warning/element_set_spec.rb` (`mkspec` can also create the file for you).

Reply:
I moved the specs to new corresponding files.